src/wp-includes/interactivity-api/class-wp-interactivity-api.php

```diff
@@ -0,0 +1,671 @@
+					$tag_stack[] = array( $tag_name, $directives_prefixes );
+				}
+			}
+
+			/*
+			 * Sorts the attributes by the order of the `directives_processor` array
+			 * and checks what directives are present in this element. The processing
+			 * order is reversed for tag closers.
+			 */
+			$directives_prefixes = array_intersect(
+				$p->is_tag_closer()
+					? $directive_processor_prefixes_reversed
+					: $directive_processor_prefixes,
+				$directives_prefixes
+			);
+
+			// Executes the directive processors present in this element.
+			foreach ( $directives_prefixes as $directive_prefix ) {
+				$func = is_array( self::$directive_processors[ $directive_prefix ] )
+					? self::$directive_processors[ $directive_prefix ]
+					: array( $this, self::$directive_processors[ $directive_prefix ] );
+				call_user_func_array(
+					$func,
+					array( $p, &$context_stack, &$namespace_stack )
+				);
+			}
+		}
+
+		/*
+		 * It returns the original content if the HTML is unbalanced because
+		 * unbalanced HTML is not safe to process. In that case, the Interactivity
+		 * API runtime will update the HTML on the client side during the hydration.
+		 */
+		return $unbalanced || 0 < count( $tag_stack ) ? $html : $p->get_updated_html();
+	}
+
+	/**
+	 * Evaluates the reference path passed to a directive based on the current
+	 * store namespace, state and context.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @param string|true $directive_value   The directive attribute value string or `true` when it's a boolean attribute.
+	 * @param string      $default_namespace The default namespace to use if none is explicitly defined in the directive
+	 *                                       value.
+	 * @param array|false $context           The current context for evaluating the directive or false if there is no
+	 *                                       context.
+	 * @return mixed|null The result of the evaluation. Null if the reference path doesn't exist.
+	 */
+	private function evaluate( $directive_value, string $default_namespace, $context = false ) {
+		list( $ns, $path ) = $this->extract_directive_value( $directive_value, $default_namespace );
+		if ( empty( $path ) ) {
+			return null;
+		}
+
+		$store = array(
+			'state'   => isset( $this->state_data[ $ns ] ) ? $this->state_data[ $ns ] : array(),
+			'context' => isset( $context[ $ns ] ) ? $context[ $ns ] : array(),
+		);
+
+		// Checks if the reference path is preceded by a negator operator (!).
+		$should_negate_value = '!' === $path[0];
+		$path                = $should_negate_value ? substr( $path, 1 ) : $path;
+
+		// Extracts the value from the store using the reference path.
+		$path_segments = explode( '.', $path );
+		$current       = $store;
+		foreach ( $path_segments as $path_segment ) {
+			if ( isset( $current[ $path_segment ] ) ) {
+				$current = $current[ $path_segment ];
+			} else {
+				return null;
+			}
+		}
+
+		// Returns the opposite if it contains a negator operator (!).
```

Comment:
```suggestion
		// Returns the opposite if it contains a negation operator (!).
```

Reply:
Thanks. Fixed in https://github.com/WordPress/wordpress-develop/pull/5953/commits/e8daa1c89140249202d9976ef73d5eef9bd78c2c.